app/components/markdown/latex_code_block/latex_code_block.js

```diff
@@ -0,0 +1,204 @@
+    });
+
+    trimContent = (content) => {
+        const lines = splitLatexCodeInLines(content);
+
+        const numberOfLines = lines.length;
+
+        if (numberOfLines > MAX_LINES) {
+            return {
+                content: lines.slice(0, MAX_LINES),
+                numberOfLines,
+            };
+        }
+
+        return {
+            content: lines,
+            numberOfLines,
+        };
+    };
+
+    onWebViewMessage = (data) => {
+        this.setState({webViewHeight: data});
+    }
+
+    render() {
+        const style = getStyleSheet(this.props.theme);
+
+        let language = null;
+        if (this.props.language) {
+            const languageDisplayName = getDisplayNameForLanguage(this.props.language);
+
+            if (languageDisplayName) {
+                language = (
+                    <View style={style.language}>
+                        <Text style={style.languageText}>
+                            {languageDisplayName}
+                        </Text>
+                    </View>
+                );
+            }
+        }
+
+        const {content, numberOfLines} = this.trimContent(this.props.content);
+
+        let plusMoreLines = null;
+        if (numberOfLines > MAX_LINES) {
+            plusMoreLines = (
+                <FormattedText
+                    style={style.plusMoreLinesText}
+                    id='mobile.markdown.code.plusMoreLines'
+                    defaultMessage='+{count, number} more {count, plural, one {line} other {lines}}'
+                    values={{
+                        count: numberOfLines - MAX_LINES,
+                    }}
+                />
+            );
+        }
+
+        /**
+         * Note on the error behavior of math view:
+         * - onError returns an Error object
+         * - renderError returns an options object with an error attribute that contains the real Error.
+         */
+        return (
+            <TouchableWithFeedback
+                onPress={this.handlePress}
+                onLongPress={this.handleLongPress}
+                type={'opacity'}
+            >
+                <View style={style.container}>
+                    <View style={style.rightColumn}>
+                        {content.map((latexCode) => (
+                            <View
+                                style={style.code}
+                                key={latexCode}
+                            >
+                                <MathView
+                                    math={latexCode}
+                                    onError={(errorMsg) => {
+                                        return <Text style={style.errorText}>{'Error: ' + errorMsg.message}</Text>;
+                                    }}
+                                    renderError={(errorMsg) => {
+                                        return <Text style={style.errorText}>{'Render error: ' + errorMsg.error.message}</Text>;
+                                    }}
```

Comment:
^ The current code makes it so that the MathView will re-render whenever the LatexCodeBlock does because it's passing a new copy of the arrow function in each time. If they're methods on the LatexCodeBlock, it'll always pass the a reference to the same method.

Reply:
Ah ok i didn't know that. I'll change that to a class method